frontend/src/scenes/session-recordings/components/AiRegexHelper/aiRegexHelperLogic.ts

```diff
@@ -0,0 +1,87 @@
+import { actions, kea, listeners, path, reducers } from 'kea'
+import api from 'lib/api'
+import { lemonToast } from 'lib/lemon-ui/LemonToast/LemonToast'
+import posthog from 'posthog-js'
+
+import type { aiRegexHelperLogicType } from './aiRegexHelperLogicType'
+
+export const aiRegexHelperLogic = kea<aiRegexHelperLogicType>([
+    path(['lib', 'components', 'AiRegexHelper', 'aiRegexHelperLogic']),
+    actions({
+        setIsOpen: (isOpen: boolean) => ({ isOpen }),
+        setInput: (input: string) => ({ input }),
+        handleGenerateRegex: () => ({}),
+        handleApplyRegex: () => ({}),
+        setIsLoading: (isLoading: boolean) => ({ isLoading }),
+        setGeneratedRegex: (generatedRegex: string) => ({ generatedRegex }),
+        setError: (error: string) => ({ error }),
+        onClose: () => ({}),
+    }),
+    reducers({
+        isOpen: [
+            false,
+            {
+                setIsOpen: (_, { isOpen }) => isOpen,
+            },
+        ],
+        input: [
+            '',
+            {
+                setInput: (_, { input }) => input,
+            },
+        ],
+        isLoading: [
+            false,
+            {
+                setIsLoading: (_, { isLoading }) => isLoading,
+            },
+        ],
+        generatedRegex: [
+            '',
+            {
+                setGeneratedRegex: (_, { generatedRegex }) => generatedRegex,
+            },
+        ],
+        error: [
+            '',
+            {
+                setError: (_, { error }) => error,
+            },
+        ],
+    }),
+    listeners(({ actions, values }) => ({
+        handleGenerateRegex: async () => {
```

Comment:
you would get some of this behaviour for free if you used a `loader` here 

https://keajs.org/docs/plugins/loaders/

not wrong to do it manually just making sure you know loaders exist :)